corehq/apps/app_manager/static/app_manager/js/form_designer.js

```diff
@@ -0,0 +1,171 @@
+                },
+            },
+            windowManager: {
+                leftOffset: function () {
+                    return $('#hq-sidebar').outerWidth() + 2;
+                },
+                topOffset: function () {
+                    return $('#hq-navigation').outerHeight();
+                },
+                toggleFullScreenCallback: function (isFullscreen) {
+                    var $preview = $('#js-appmanager-preview');
+                    if (isFullscreen) {
+                        $preview.addClass('fullscreen');
+                    } else {
+                        $preview.removeClass('fullscreen');
+                    }
+                },
+            },
+            csrftoken: $.cookie('csrftoken'),
+        });
+
+        // Add callbacks to core, which has already been provided by the server
+        if (v2) {
+            VELLUM_OPTIONS.core = _.extend(VELLUM_OPTIONS.core, {
+                formLoadingCallback: function() {
+                    $('#formdesigner').addClass('loading');
+                },
+                formLoadedCallback: function() {
+                    $('#formdesigner').removeClass('loading');
+                    $('#formdesigner .fd-content-left .fd-head-text').after(
+                        $('#fd-hq-edit-formname-button').html()
+                    );
+                },
+            });
+        }
+        VELLUM_OPTIONS.core = _.extend(VELLUM_OPTIONS.core, {
+            onFormSave: function(data) {
+                var app_manager = hqImport('app_manager/js/app_manager.js');
+                app_manager.updateDOM(data.update);
+                if (v2) {
+                    $('.js-preview-toggle').removeAttr('disabled');
+                }
+                if (initial_page_data("days_since_created")) {
+                    analytics.workflow('Saved the Form Builder within first 24 hours');
+                }
+            },
+            onReady: function() {
+                if (initial_page_data('vellum_debug') === 'dev') {
+                    var less_error_id = "#less-error-message\\:static-style-less-hqstyle-core",
+                        less_error = $(less_error_id);
+                    if (less_error.length) {
+                        console.log("hiding less error:", less_error_id);
+                        console.log(less_error.text());
+                        less_error.hide();
+                    }
+                }
+                if (initial_page_data('guided_tour')) {
+                    form_tour_start();
+                }
+                if (initial_page_data('days_since_created') === 0) {
+                    $("#formdesigner").vellum("get").data.core.form.on("question-create", function() {
+                        analytics.workflow('Added question in Form Builder within first 24 hours');
+                    });
+                }
+            },
+        });
+
+        var CKEDITOR_BASEPATH = initial_page_data('CKEDITOR_BASEPATH');
```

Comment:
this is not a global because you're using `var`, might break whatever is depending on having this set